Adding authentication_keys option to os_volume EP

petr-balogh opened the pull request:
PR in ansible:
https://github.com/ansible/ansible/pull/40294

petr-balogh committed:
Adding authentication_keys option to os_volume EP

PR in ansible:
https://github.com/ansible/ansible/pull/40294

mwperina reviewed (commented):
Shouldn't we include in the role updated module once it gets merged to Ansible 2.6, so this is still usable with Ansible 2.5?

petr-balogh commented:
Thx @mwperina I can include module from my PR in ansible, but last time I did it, it didn't work for us because we ran multiple roles i our playbook, and when ansible once load module from different role, it won't reload it from this one (was our case).  So I don't mind to wait with this PR till https://github.com/ansible/ansible/pull/40294 will be merged and 2.6 is out. So should I Include this module (Still waiting if I will get some comments, so once it will be approved I can update here)?

mwperina commented:
> Thx @mwperina I can include module from my PR in ansible, but last time I did it, it didn't work for us because we ran multiple roles i our playbook, and when ansible once load module from different role, it won't reload it from this one (was our case). So I don't mind to wait with this PR till ansible/ansible#40294 will be merged and 2.6 is out. So should I Include this module (Still waiting if I will get some comments, so once it will be approved I can update here)?

We can't do anything with this bug, but we need to make sure the works fine with Ansible 2.5 when executed standalone. So please add the module with a name ovirt_external_provider_26.py so we know the module can be removed when we bump ansible version requirement for the role

petr-balogh committed:
External_provider: updated module from ansible.

Module taken from PR https://github.com/ansible/ansible/pull/40294

mwperina commented on library/README.md:
```diff
@@ -0,0 +1,10 @@
+
+Here is the list of updated modules:
+
+1. ovirt_external_provider.py
```
Please rename to `ovirt_extenal_provider_26.py` here

petr-balogh replied:
done

petr-balogh committed:
ovirt_extenal_provider.py renamed with _26 suffix

petr-balogh commented:
Hi, @mwperina I fixed requested change. Thx

mwperina commented on roles/oVirt.external-providers/tasks/main.yml:
```diff
@@ -1,6 +1,6 @@
 ---
 - name: Add external providers
-  ovirt_external_providers:
+  ovirt_external_provider:
```
You need to replace all invocation of `ovirt_external_provider` inside the role with `ovirt_external_provider_26`, otherwise the updated module won't be used

petr-balogh committed:
Remove module from ansible 2.6

We need to merge this change right after 2.6 is released because of
storage connection id change from list to string.

petr-balogh commented:
@mwperina after discussion with @machacekondra we decided to merge this change after 2.6 will be released because there is change with id in storage domains and this won't work with ansible 2.5 cause of this change. So I removed module from this PR.  Going to verify it with our infra now.. will let you know if it's working ok now later on.

petr-balogh commented:
Hey, Ansible 2.6 is out. Can this change be merged and can you make new release with this change? Thx

mwperina commented:
> Hey, Ansible 2.6 is out. Can this change be merged and can you make new release with this change? Thx

We haven't bumped requirement to Ansible 2.6, all oVirt roles still requires 2.5+, so you need to include  ovirt_external_provider from Ansible 2.6, rename it to ovirt_external_provider_26 and update all references to it within the role.

Once we will decide to bump oVirt roles requirement to ansible 2.6 we will remove the module and change source code, but for now we are still supporting ansible 2.5+

petr-balogh committed:
Added back the module from 2.6

Cause it doesn't that roles will bump to 2.6 soon, I am adding back
updated module from 2.6 .

petr-balogh commented:
Thanks, verified in our infra.

On Tue, Aug 14, 2018, 12:52 Martin Perina <notifications@github.com> wrote:

> *@mwperina* approved this pull request.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/oVirt/ovirt-ansible-infra/pull/33#pullrequestreview-146010649>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/APa6kQVM4LgSzOILppINjiVHxdoM1HSJks5uQqwBgaJpZM4UCMlZ>
> .
>

petr-balogh commented:
Hi @machacekondra  can we merge this one? Thanks
